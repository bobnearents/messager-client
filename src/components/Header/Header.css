@import url('https://fonts.googleapis.com/css?family=Pacifico');


.Header {
  background:#264653;
  min-height:60px;
  height:10vh;
  display:flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  padding:10px;
  color:#CFDBD5;
  z-index: 19;
}

.Header * {
  margin-right: 10px;
  font-size: 1.3em;
}

.logo {
  font-family: 'Pacifico';
  font-size:1.9em;
  z-index: 20;
  align-self: flex-start;
}

.logo a {
  color:#D5EDE1;
  text-decoration: none;
}

.toggle-rooms-bars {
  display:none;
}


@media(max-width:668px) {

  .toggle-rooms-bars {
    font-size: 1.5em;
    display:inline;
  }
  
}
