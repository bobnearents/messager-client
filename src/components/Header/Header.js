import React, { Component } from "react";
import "./Header.css";
import MessageContext from "../../context/message-context";
import { Link } from "react-router-dom";
import { FaBars, FaTimes } from "react-icons/fa";
import TokenService from '../../services/token-service';

export default class Header extends Component {
  static contextType = MessageContext;

  renderIcon = () => {
    if(TokenService.hasAuthToken()) {
      if (this.context.isRoomActive) {
      return <FaTimes 
        className="toggle-rooms-bars"
        onClick={() => {
        this.context.toggleActiveRoom() }} 
      />
    }
    return <FaBars 
      className="toggle-rooms-bars"
      onClick={() => {
        this.context.toggleActiveRoom()
      }}
    />}
  }

  render() {
    return (
      <>
        <nav className="Header">
          <span className='logo'><Link to='/'>messager.</Link></span>
          <span className = 'user-greeting'>
          greetings {this.context.user}!</span>
          {this.renderIcon()}
        </nav>
      </>
    );
  }
}
