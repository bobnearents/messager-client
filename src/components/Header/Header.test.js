import React from 'react';
import { shallow } from 'enzyme';
import renderer from "react-test-renderer";
import { Route, Link, MemoryRouter } from "react-router-dom";


import Header from './Header';

describe('<Header />', () => {

  it('Renders without crashing', () => {
    shallow(<Header />)
  })

  it("renders correctly", () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Header />
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
})