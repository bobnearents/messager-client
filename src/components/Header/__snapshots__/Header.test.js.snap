// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders correctly 1`] = `
<nav
  className="Header"
>
  <span
    className="logo"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      messager.
    </a>
  </span>
  <span
    className="user-greeting"
  >
    greetings 
    !
  </span>
</nav>
`;
